packages/features/auth/SAMLLogin.tsx

```diff
@@ -21,10 +22,14 @@ const schema = z.object({
 export function SAMLLogin({ samlTenantID, samlProductID, setErrorMessage }: Props) {
   const { t } = useLocale();
   const methods = useFormContext();
+  const [lastUsed, setLastUsed] = useLastUsed();
 
   const mutation = trpc.viewer.public.samlTenantProduct.useMutation({
     onSuccess: async (data) => {
-      await signIn("saml", {}, { tenant: data.tenant, product: data.product });
+      const res = await signIn("saml", {}, { tenant: data.tenant, product: data.product });
+      if (res && !res.error) {
+        setLastUsed("saml");
```

Comment:
I think it would make more practical if setLastUsed was only used inside of login.tsx, but in the case this would require passing an onSuccess prop to SAMLLogin